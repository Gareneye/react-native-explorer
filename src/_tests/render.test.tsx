import * as React from "react"
import { View } from "react-native"
import * as renderer from "react-test-renderer"

import componentTreeToNodeTree from "../component-tree-to-nodes"
import treeToSVG from "../tree-to-svg"

import * as fs from "fs"
import * as yoga from "yoga-layout"

it("handles some simple JSX", () => {
  const jsx = (
    <View style={{width: 200, height: 200, marginLeft: 40, marginTop: 40, paddingLeft: 20}}>
      <View style={{width: 20, height: 20}}/>
      <View style={{width: 20, height: 20}}/>
      <View style={{width: 20, height: 20}}/>
    </View>
  )
  const component = renderer.create(jsx).toJSON()
  const settings = {
    width:  600,
    height: 400,
  }
  console.log(component)
  const rootNode = componentTreeToNodeTree(component, settings)
  const results = treeToSVG(rootNode, settings)

  fs.writeFileSync("jsx-render.svg", results)
  expect(results).toMatchSnapshot()

  rootNode.freeRecursive()
})
