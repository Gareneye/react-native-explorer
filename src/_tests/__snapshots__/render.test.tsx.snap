// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles some simple JSX 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>
<svg width=\\"600\\" height=\\"400\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\">

 <rect x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\" fill-opacity=\\"0.05\\" stroke-width=\\"1\\" stroke=\\"black\\"/>

 <g transform='translate(0, 0)'>
   <rect x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\" fill-opacity=\\"0.05\\" stroke-width=\\"1\\" stroke=\\"black\\"/>
   <rect x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\" fill-opacity=\\"0.05\\" stroke-width=\\"1\\" stroke=\\"black\\"/>

   <g transform='translate(0, 0)'>
     <rect x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\" fill-opacity=\\"0.05\\" stroke-width=\\"1\\" stroke=\\"black\\"/>
     <rect x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\" fill-opacity=\\"0.05\\" stroke-width=\\"1\\" stroke=\\"black\\"/>
   </g>

   <rect x=\\"0\\" y=\\"0\\" width=\\"NaN\\" height=\\"NaN\\" fill-opacity=\\"0.05\\" stroke-width=\\"1\\" stroke=\\"black\\"/>
 </g>

</svg>
"
`;
